yogstation/code/modules/mob/living/carbon/human/species_types/mothmen.dm

```diff
@@ -0,0 +1,26 @@
+/datum/species/moth
+	var/MothLightRange = 15 //In what radius shall moths look for lights to fly at?
+	var/MothLightProb = 2 //Don't want them doing it tooooo often..Unless an admin changes it :^) So change this back to 5 after testing
+
+/datum/species/moth/spec_life(mob/living/carbon/human/H)
+	. = ..()
+	if(H.stat != DEAD) //Suffer not a dead moth to move
+		MothLightProb = initial(MothLightProb)
+		var/retarded = H.getBrainLoss()
+		MothLightProb += retarded*0.3 //So there's always a slight chance.
+		var/list/LightsIWantToFlyAt = list()
+		if(prob(MothLightProb)) //Moth people now have a random urge to fly violently into lights
+			for(var/obj/O in orange(H,MothLightRange))
```

Comment:
im not great at optimization but orange in a 15 range on life sounds like a bit of a high performance task for something as memey at this